packages/glob/src/internal-pattern.ts

```diff
@@ -48,8 +48,13 @@ export class Pattern {
   // https://github.com/typescript-eslint/typescript-eslint/issues/291
 
   constructor(pattern: string)
+  constructor(pattern: string, segments: undefined, homedir: string)
```

Comment:
all the changes to this file are reaction due to moving `"esModuleInterop": true` to the root `tsconfig.json` - which previously was only defined in `packages/github/tsconfig.json`

mocking `os` in the module registry from the unit tests, stopped working when esModuleInterop=true so i switched to a constructor overload pattern